desktop/app/src/ui/components/data-inspector/DataDescription.tsx

```diff
@@ -596,17 +596,22 @@ class DataDescriptionContainer extends Component<{
 
       case 'text':
       case 'string':
-        if (val.startsWith('http://') || val.startsWith('https://')) {
+        const isUrl = val.startsWith('http://') || val.startsWith('https://');
+        const editIcon = (
+          <Glyph
+            name="pencil"
+            variant="outline"
+            color={colors.light20}
+            size={16}
+            style={{cursor: 'pointer', marginLeft: 8}}
+          />
+        );
+
+        if (isUrl) {
           return (
             <>
               <Link href={val}>{val}</Link>
-              <Glyph
-                name="pencil"
-                variant="outline"
-                color={colors.light20}
-                size={16}
-                style={{cursor: 'pointer', marginLeft: 8}}
-              />
+              {editable && editIcon}
```

Comment:
Thanks for this PR! Would you mind moving the element construction back in to the last block? With the current setup, the element will be instantiated for every string that is rendered, and not used in most cases (e.g. when it i not an url, or when the url is not editable)

Reply:
Makes sense, good catch! Refactored in the latest commit